eclair-core/src/test/scala/fr/acinq/eclair/api/JsonRpcServiceSpec.scala

```diff
@@ -0,0 +1,270 @@
+        val peerInfos = response.result.asInstanceOf[Seq[Map[String,String]]]
+        assert(peerInfos.size == 2)
+        assert(peerInfos.head.get("nodeId") == Some(Alice.nodeParams.nodeId.toString))
+        assert(peerInfos.head.get("state") == Some("CONNECTED"))
+        matchTestJson("peers", false, response)
+      }
+  }
+
+  test("'getinfo' response should include this node ID") {
+    val mockService = new {} with MockService {
+      override def getInfoResponse: Future[GetInfoResponse] = Future.successful(GetInfoResponse(
+        nodeId = Alice.nodeParams.nodeId,
+        alias = Alice.nodeParams.alias,
+        port = 9735,
+        chainHash = Alice.nodeParams.chainHash,
+        blockHeight = 123456
+      ))
+    }
+    import mockService.formats
+    import mockService.serialization
+
+    val postBody = JsonRPCBody(method = "getinfo", params = Seq.empty)
+
+    Post("/", postBody) ~>
+      addCredentials(BasicHttpCredentials("", mockService.password)) ~>
+      addHeader("Content-Type", "application/json") ~>
+      Route.seal(mockService.route) ~>
+      check {
+        assert(handled)
+        assert(status == OK)
+        val resp = entityAs[JsonRPCRes]
+        assert(resp.result.toString.contains(Alice.nodeParams.nodeId.toString))
+        matchTestJson("getinfo", false ,resp)
+      }
+  }
+
+  test("'close' method should accept a shortChannelId") {
+
+    val shortChannelIdSerialized = "42000x27x3"
+
+    val mockService = new MockService(defaultMockKit.copy(
+      register = system.actorOf(Props(new {} with MockActor {
+        override def receive = {
+          case ForwardShortId(shortChannelId, _) if shortChannelId.toString == shortChannelIdSerialized =>
+            sender() ! Alice.nodeParams.nodeId.toString
+        }
+      }))
+    ))
+
+    import mockService.formats
+    import mockService.serialization
+
+
+    val postBody = JsonRPCBody(method = "close", params = Seq(JString(shortChannelIdSerialized)))
+
+    Post("/", postBody) ~>
+      addCredentials(BasicHttpCredentials("", mockService.password)) ~>
+      addHeader("Content-Type", "application/json") ~>
+      Route.seal(mockService.route) ~>
+      check {
+        assert(handled)
+        assert(status == OK)
+        val resp = entityAs[JsonRPCRes]
+        assert(resp.result.toString.contains(Alice.nodeParams.nodeId.toString))
+        matchTestJson("close", false ,resp)
+      }
+  }
+
+  private def readFileAsString(path: Path): String = Files.exists(path) match {
+    case true => new String(Files.readAllBytes(path.toAbsolutePath))
+    case false => throw new IllegalArgumentException(s"Mock file for $path does not exist, please use 'overWrite' first.")
+  }
+
+  private def matchTestJson(rpcMethod: String, overWrite: Boolean, response: JsonRPCRes)(implicit formats: Formats) = {
+    val responseContent = Serialization.writePretty(response)
+    val path = Paths.get(s"src/test/resources/api/$rpcMethod")
```

Comment:
This doesn't work from intellij because you are accessing resources using a relative path.

Workaround (quite dirty, I would cleanly separate the "overwrite" part):
```diff
diff --git a/eclair-core/src/test/scala/fr/acinq/eclair/api/JsonRpcServiceSpec.scala b/eclair-core/src/test/scala/fr/acinq/eclair/api/JsonRpcServiceSpec.scala
index 86f2bc3a..e92308ad 100644
--- a/eclair-core/src/test/scala/fr/acinq/eclair/api/JsonRpcServiceSpec.scala
+++ b/eclair-core/src/test/scala/fr/acinq/eclair/api/JsonRpcServiceSpec.scala
@@ -1,9 +1,11 @@
 package fr.acinq.eclair.api


+import java.io.InputStream
 import java.nio.charset.StandardCharsets._
 import java.nio.file.{Files, Path, Paths, StandardOpenOption}
 import java.nio.file.StandardOpenOption._
+
 import akka.actor.{Actor, ActorSystem, Props, Scheduler}
 import org.scalatest.FunSuite
 import akka.http.scaladsl.model.StatusCodes._
@@ -25,6 +27,8 @@ import org.json4s.jackson.Serialization

 import scala.concurrent.Future
 import scala.concurrent.duration._
+import scala.io.Source
+import scala.util.Try

 class JsonRpcServiceSpec extends FunSuite with ScalatestRouteTest {

@@ -248,19 +252,18 @@ class JsonRpcServiceSpec extends FunSuite with ScalatestRouteTest {
       }
   }

-  private def readFileAsString(path: Path): String = Files.exists(path) match {
-    case true => new String(Files.readAllBytes(path.toAbsolutePath))
-    case false => throw new IllegalArgumentException(s"Mock file for $path does not exist, please use 'overWrite' first.")
-  }
+  private def readFileAsString(stream: InputStream): Try[String] = Try(Source.fromInputStream(stream).mkString)

   private def matchTestJson(rpcMethod: String, overWrite: Boolean, response: JsonRPCRes)(implicit formats: Formats) = {
     val responseContent = Serialization.writePretty(response)
-    val path = Paths.get(s"src/test/resources/api/$rpcMethod")
+    val fileName = s"/api/$rpcMethod"
     if(overWrite){
+      val path = Paths.get(s"src/test/resources$fileName")
       Files.write(path, responseContent.getBytes(UTF_8), TRUNCATE_EXISTING, CREATE)
       assert(false, "'overWrite' should be false before commit")
     }else{
-      val expectedResponse = readFileAsString(path)
+      val stream = getClass.getResourceAsStream(fileName)
+      val expectedResponse = readFileAsString(stream).getOrElse(throw new IllegalArgumentException(s"Mock file for $fileName does not exist, please use 'overWrite' first."))
       assert(responseContent == expectedResponse, s"test mock for $rpcMethod did not match the expected response")
     }
```

Reply:
Commit 01ef32d makes it possible to read/write from the test even when run from intellij